internal/storage/bucket_handle.go

```diff
@@ -64,6 +64,10 @@ func (bh *bucketHandle) NewReader(
 		obj = obj.Generation(req.Generation)
 	}
 
+	if req.ReadCompressed {
```

Comment:
It should have unit tests? in bucket_handle_test ?

Reply:
Added. Please check.